### PR TITLE
single interest factory per net driver

### PR DESCRIPTION
**Pull request by jacquese**
general refactor to have a single interest factory which stores state to be reused throughout the lifetime of a single "run" (whatever the net driver's lifetime is defined by)

adds some comments to describe the API of the class and its responsibilities

will drag some things out into their own classes in order to improve testability and then add unit tests in future PRs

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Public/Utils/SpatialInterestFactory.h [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+class UAbstractLBStrategy;
+class USpatialClassInfoManager;
+class USpatialPackageMapClient;
+
+DECLARE_LOG_CATEGORY_EXTERN(LogInterestFactory, Log, All);
+
+namespace SpatialGDK
+{
+class SPATIALGDK_API SpatialInterestFactory
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Public/Utils/SpatialInterestFactory.h]: Lets just call it `InterestFactory`. We generally only add Spatial to things that have a native version.

File: SpatialGDK/Source/SpatialGDK/Public/Schema/Interest.h [hunk 2]
```diff
@@ -7,6 +7,7 @@
 namespace SpatialGDK
 {
 using EdgeLength = Coordinates;
+using ResultType = TArray<Worker_ComponentId>;
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Public/Schema/Interest.h]: Didn't this already go in?

Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Public/Schema/Interest.h]: I think it got lost in one of the rebase races

File: SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp [hunk 3]
```diff
@@ -29,28 +29,26 @@ struct FrequencyConstraint
 	SpatialGDK::QueryConstraint Constraint;
 };
 // Used to cache checkout radius constraints with frequency settings, so queries can be quickly recreated.
-static TArray<FrequencyConstraint> CheckoutConstraints;
+TArray<FrequencyConstraint> CheckoutConstraints;
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp]: These should be member variables now

File: SpatialGDK/Source/SpatialGDK/Public/Utils/InterestFactory.h [hunk 4]
```diff
@@ -7,53 +7,75 @@
...
+ *
+ * The first is actor interest. The factory takes information about an actor (the object, info and corresponding entity ID)
+ * and produces an interest data/update for that entity. This interest contains anything specific to that actor, such as self constraints
+ * for servers and clients, and if the actor is a player controller, the client worker's interest is also built for that actor.
+ *
+ * The other is server worker interest. Given a load balancing strategy, the factory will take the strategy's defined query constraint
+ * and produce an interest component to exist on the server's worker entity. This interest component contains the primary interest query made
+ * by that server worker.
+ */
+
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Public/Utils/InterestFactory.h]: Nice

File: SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp (hunk 3 above)
Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp]: These are member variables now

File: SpatialGDK/Source/SpatialGDK/Public/Utils/InterestFactory.h (hunk 4 above)
Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Public/Utils/InterestFactory.h]: ty

File: SpatialGDK/Source/SpatialGDK/Public/Utils/InterestFactory.h [hunk 5]
```diff
@@ -7,53 +7,82 @@
...
+	// TODO: Pull out into checkout radius constraint utils
+	QueryConstraint CreateClientCheckoutRadiusConstraint(USpatialClassInfoManager* ClassInfoManager);
 	static QueryConstraint CreateLegacyNetCullDistanceConstraint(USpatialClassInfoManager* ClassInfoManager);
 	static QueryConstraint CreateNetCullDistanceConstraint(USpatialClassInfoManager* ClassInfoManager);
-	static QueryConstraint CreateNetCullDistanceConstraintWithFrequency(USpatialClassInfoManager* ClassInfoManager);
+	QueryConstraint CreateNetCullDistanceConstraintWithFrequency(USpatialClassInfoManager* ClassInfoManager);
 
 	// Builds the result types of necessary components for clients
+	// TODO: create and pull out into result types class
 	static ResultType CreateClientNonAuthInterestResultType(USpatialClassInfoManager* ClassInfoManager);
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Public/Utils/InterestFactory.h]: None of these functions need to be static

Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Public/Utils/InterestFactory.h]: done

**Pull request merged**